src/analysis/inner_product_space/pi_L2.lean

```diff
@@ -164,6 +164,24 @@ end
 
 end
 
+/-- The vector given in euclidean space by being `1 : 𝕜` at coordinate `i : ι` and `0 : 𝕜` at
+all other coordinates. -/
+def euclidean_space.projₗ [decidable_eq ι] (i : ι) :
+  euclidean_space 𝕜 ι →ₗ[𝕜] 𝕜 :=
+(linear_map.proj i).comp (pi_Lp.linear_equiv 2 𝕜 (λ i : ι, 𝕜) : euclidean_space 𝕜 ι →ₗ[𝕜] ι → 𝕜)
+
+/-- The vector given in euclidean space by being `1 : 𝕜` at coordinate `i : ι` and `0 : 𝕜` at
+all other coordinates. -/
+def euclidean_space.proj [decidable_eq ι] (i : ι) :
+  euclidean_space 𝕜 ι →L[𝕜] 𝕜 :=
+⟨euclidean_space.projₗ i, continuous_apply i⟩
+
+@[simp] lemma euclidean_space.coe_proj [decidable_eq ι] (i : ι) :
+  ⇑(euclidean_space.proj i : euclidean_space 𝕜 ι →L[𝕜] 𝕜) = function.eval i := rfl
```

Comment:
This lemma might be a bad idea, as it encourages terms of type `euclidean_space 𝕜 ι` to decay to `ι → 𝕜`. The one below is fine.